### DISCUSSION TITLE
16-bit parallel with the ILI9488 and ESP32-S3

### DISCUSSION THREAD

**Discussion by pastaskrue** [Bodmer/TFT_eSPI#2387] [Compatible displays and setup files]
Hello there!

I have been looking around for a tutorial (or explanation) on how to connect the ESP32-S3 to a display (ILI9488 driver) in the 16-bit parallel configuration.

Can I just connect them as shown in this setup file? Or do I have to use other pin connections?
https://github.com/Bodmer/TFT_eSPI/blob/master/User_Setups/Setup107_RP2040_ILI9341_16bit_parallel.h

```cpp
// These pins can be moved and are controlled directly by the library software
#define TFT_RST   18  // Reset pin
#define TFT_CS    19  // Do not define if chip select control pin permanently connected to 0V
//#define TFT_RD   -1   // Do not define, read pin must be permanently connected to 3V3

// Note: All the following pins are PIO hardware configured and driven
#define TFT_WR   16    // Write strobe pin
#define TFT_DC   17    // Data Command control pin

 // PIO requires these to be sequentially increasing
#define TFT_D0    0
#define TFT_D1    1
#define TFT_D2    2
#define TFT_D3    3
#define TFT_D4    4
#define TFT_D5    5
#define TFT_D6    6
#define TFT_D7    7
#define TFT_D8    8
#define TFT_D9    9
#define TFT_D10  10
#define TFT_D11  11
#define TFT_D12  12
#define TFT_D13  13
#define TFT_D14  14
#define TFT_D15  15
```

And is there an explanation of why PlatformIO wants the pins to be sequentially increased like shown above? Saw someone using the [LovyanGFX](https://github.com/Makerfabs/Makerfabs-ESP32-S3-Parallel-TFT-with-Touch/blob/main/firmware/SD16_3.5/SD16_3.5.ino) library but not with sequentially increasing pin numbers.

If anyone has the pin connections from the ESP32-S3 to the display (ILI9488) in 16-bit parallel laying around and is willing to show, share or tell how they did it - would be greatly appreciated!

Best regards
Ask

**Comment by Bodmer** [comment]
The library only supports 8 bit parallel interfaces with the ESP32 family. The 16 bit interface is only supported with RP2040 processors (e.g. Raspberry Pico).

**Comment by paulw517** [comment]
Hello Bodmer, I am confused about the 8bit and 16bit issue above, 

Makerfabs say their  **MaTouch_ESP32-S3 Parallel TFT with Touch 3.5'' ILI9488** is a 16bit utilizing an **ESP32-S3-WROOM-1-N16R2** micro???

I am attempting to pass live, real-time GPS data from an TTGO T7 V1.3 ESP32-WROOM to the Makerfabs **ESP32-S3_Wroom equipped** ILI9488 LCD Display (referenced below) via I2C communication.  However I am not able to achieve communication beyond basic chip info, attachment **I2C does not receive data.png.**

https://www.makerfabs.com/esp32-s3-parallel-tft-with-touch-ili9488.html
SKU ESP32S335D

Controller: ESP32-S3-WROOM-1, PCB Antenna, 16MB Flash, 2MB PSRAM, ESP32-S3-WROOM-1-N16R2
Wireless: Wifi& Bluetooth 5.0
LCD: 3.5inch TFT LCD
Resolution: 480*320
Color: RGB
**LCD interface: 16bit parallel**
LCD Driver: ili9488



![I2C correctly passing GPS data](https://github.com/Bodmer/TFT_eSPI/assets/11984841/02f384eb-48a8-41c4-8350-f9115ffc7099)
![I2C does not receive data](https://github.com/Bodmer/TFT_eSPI/assets/11984841/15d91d11-3960-461b-87d6-0a6bb0600d40)
llel**
